tests/integration-tests/tests/createami/test_createami.py

```diff
@@ -116,8 +107,74 @@ def test_createami_post_install(
         + post_install_args
     )
 
-    assert_that(
-        any("No post install script".lower() in s.lower() for s in pcluster_createami_result.stdout.split("\n"))
-    ).is_false()
+    stdout_lower = pcluster_createami_result.stdout.lower()
+    assert_that(stdout_lower).does_not_contain("no post install script")
+    assert_that(stdout_lower).does_not_contain("no custom ami created")
+
+
+@pytest.mark.dimensions("eu-central-1", "c5.xlarge", "alinux", "sge")
+def test_createami_wrong_os(
+    region, os, instance, test_datadir, request, pcluster_config_reader, vpc_stack, architecture, scheduler
+):
+    """Test error message when os provide is different from the os of custom AMI"""
+    cluster_config = pcluster_config_reader()
+
+    # alinux is specified in the config file but an AMI of alinux2 is provided
+    wrong_os = "alinux2"
+    logging.info("Asserting os fixture is different from wrong_os variable")
+    assert_that(os != wrong_os).is_true()
+    base_ami = retrieve_latest_ami(region, wrong_os, ami_type="pcluster", architecture=architecture)
+
+    # Networking
+    vpc_id = vpc_stack.cfn_outputs["VpcId"]
+    subnet_id = vpc_stack.cfn_outputs["PublicSubnetId"]
+    networking_args = f"--vpc-id {vpc_id} --subnet-id {subnet_id}"
+
+    # Custom Cookbook
+    custom_cookbook = request.config.getoption("createami_custom_chef_cookbook")
+    custom_cookbook_args = "" if not custom_cookbook else f"-cc {custom_cookbook}"
+
+    command = (
+        f"pcluster createami -ai {base_ami} -os {os} -r {region} -c {cluster_config.as_posix()} "
+        f"{custom_cookbook_args} -i {instance} {networking_args}"
+    )
+    createami_process = pexpect.spawn(command, timeout=600)
+    logging.info("Verifying errors in stdout")
+    createami_process.expect("RuntimeError")
+    createami_process.expect(fr"custom AMI.+{wrong_os}.+base.+os.+config file.+{os}")
+    createami_process.expect("No custom AMI created")
+    createami_process.close()
+
+
+@pytest.mark.dimensions("ca-central-1", "c5.xlarge", "alinux", "sge")
+def test_createami_wrong_pcluster_version(
+    region, os, instance, test_datadir, request, pcluster_config_reader, vpc_stack, architecture, scheduler
```

Comment:
`scheduler` is not used in the test, so probably don't need to pass it in?
Please double check for all fixtures, I think `test_datadir ` and `architecture` are not used for this particular test as well?

Reply:
Done